src/main/java/dude/task/Deadline.java

```diff
@@ -0,0 +1,43 @@
+package dude.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task with a deadline in Dude.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
+
+    /**
+     * Constructs a Deadline with the specified description and deadline.
+     * Date and time must be in specific format: "yyyy-MM-dd HH:mm"
+     *
+     * @param description The description of the task.
+     * @param by The date and time represent the deadline.
+     */
```

Comment:
I really like how you clearly explain what the constructor does in each of your classes.